etc/generate-latest-apidocs.sh

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env bash
+#
+# generate-latest-apidocs.sh
+#
+# Usage:
+#   ./etc/generate-latest-apidocs.sh
+#
+# This script is meant to be run from the project root directory.
+
+set -o errexit
+set -o pipefail
+
+use_current="no"
+
+while [[ "$#" > 0 ]]; do
+  case "${1:?}" in
+  --current)
+    use_current="yes"
+    ;;
+  *)
+    echo "unrecognized argument: ${1:?}" 1>&2
+    exit 1
+    ;;
+  esac
+
+  shift
+done
+
+LATEST_VERSION="4.0.0"
+DOXYGEN_VERSION_REQUIRED="1.12.0"
+
+# Permit using a custom Doxygen binary.
+: "${DOXYGEN_BINARY:=doxygen}"
+
+command -v git >/dev/null
+command -v mktemp >/dev/null
+command -v sed >/dev/null
+
+if [[ ! -d build ]]; then
+  echo "missing build directory: this script must be run from the project root directory" 1>&2
+  exit 1
+fi
+
+# Validate Doxygen version.
+doxygen_version="$("${DOXYGEN_BINARY:?}" -v | perl -lne 'print $1 if m/^(\d+\.\d+\.\d+).*$/')"
+if [[ "${doxygen_version:-}" != "${DOXYGEN_VERSION_REQUIRED:?}" ]]; then
+  echo "${DOXYGEN_BINARY} version ${doxygen_version:-"unknown"} does not equal ${DOXYGEN_VERSION_REQUIRED:?}" 1>&2
+  exit 1
```

Comment:
Requiring a version of Doxygen appears to cause failures in the [packaging tasks](https://spruce.mongodb.com/task/mongo_cxx_driver_packaging_packaging_debian_patch_5079fccebbae94bf5d3c53444acebaaed5ff8996_6762ed585a0e580007f6b323_24_12_18_15_42_21/logs?execution=0):

> doxygen version 1.9.8 does not equal 1.12.0

The Debian package appears to include docs built with the version of Doxygen available from apt. Consider reducing this to a warning for the `doxygen-current` target.

Reply:
@rcsanchez97 The [override_dh_auto_build](https://github.com/mongodb/mongo-cxx-driver/blob/debian/unstable/debian/rules#L30C1-L30C23) rule in the debian/unstable branch (along with other related scripts) may need to be updated to set `DOXYGEN_ANY_VERSION=1` when building the `doxygen-current` target. Alternatively they may be updated to avoid building API docs if that is permitted.

Reply:
@eramongodb Thanks for the heads up. The solution was fairly trivial, but testing it was a bit of a different story (owing to the Debian packaging being on a separate branch). I figured out a way to inject the proposed change into a patch build and confirmed that it works ([here](https://spruce.mongodb.com/task/mongo_cxx_driver_packaging_packaging_debian_patch_f1ad6b63d33672b6875e096765ee782046b515a8_676afb9a00bb6d00077bef4f_24_12_24_18_21_15/logs?execution=0)).

After pushing the tested change (on the `debian/unstable` branch), I re-ran the task on the waterfall and it worked ([here](https://evergreen.mongodb.com/task/mongo_cxx_driver_packaging_packaging_debian_f1ad6b63d33672b6875e096765ee782046b515a8_24_12_20_18_05_45)).